library/lua/test/expect.lua

```diff
@@ -101,25 +101,130 @@ function expect.table_eq(a, b, comment)
             ('key %s: "%s" ~= "%s"'):format(keystr, diff[1], diff[2])
 end
 
-function expect.error(func, ...)
-    local ok, ret = pcall(func, ...)
+function expect.error(func, comment)
+    local ok = pcall(func)
     if ok then
-        return false, 'no error raised by function call'
+        return false, comment, 'no error raised by function call'
     else
         return true
     end
 end
 
-function expect.error_match(func, matcher, ...)
-    local ok, err = pcall(func, ...)
-    if ok then
-        return false, 'no error raised by function call'
+local function matches(obj, matcher)
+    if not matcher then return false end
+    if type(matcher) == 'boolean' then
+        return true
     elseif type(matcher) == 'string' then
-        if not tostring(err):match(matcher) then
-            return false, ('error "%s" did not match "%s"'):format(err, matcher)
+        return tostring(obj):match(matcher)
+    elseif type(matcher) == 'function' then
+        return matcher(obj)
+    end
+    return false
+end
+
+-- matches errors thrown from the specified function. the check passes if an
+-- error is thrown and the thrown error matches the specified matcher.
+--
+-- matcher can be:
+--  a string, interpreted as a lua pattern that matches the error message
+--  a function that takes an err object and returns a boolean (true means match)
+--  the literal value true, which matches any thrown error
+function expect.error_match(matcher, func, comment)
```

Comment:
I might have missed this earlier - I see this changed from `func, matcher, comment` to `matcher, func, comment`. Is that intended to match the `expected, actual, comment` API of some of the other checks, like `expect.eq()`? (I know it's meant to match the API of `expect.printerr_match()`). Totally fine with me to standardize on something; I don't think I really considered this earlier, and making this change now with limited callers is better than making it later.

Reply:
that's right, it's now `expected, actual, comment`, but it wasn't just a swap of the first two params. The original `error_match` was `func, matcher, func_args..` with no provision for a comment.